app/client/components/homePage/banner/Countdown.tsx

```diff
@@ -0,0 +1,63 @@
+import React, { CSSProperties, useEffect } from "react";
+
+interface CountdownProps {
+  endDate: Date;
+  onCountdownFinish: () => void;
+}
+
+interface CountdownSpanCSS extends CSSProperties {
+  "--value": number;
+}
+
+export const Countdown = ({ endDate, onCountdownFinish }: CountdownProps) => {
+  const calculateTimeLeft = () => {
+    const currentDate = new Date();
+    const milisDifference = +endDate - +currentDate;
+
+    let timeLeft = {
+      days: 0,
+      hours: 0,
+      minutes: 0,
+      seconds: 0,
+    };
+
+    if (milisDifference > 0) {
+      const seconds = Math.floor(milisDifference / 1000);
+      const minutes = Math.floor(seconds / 60);
+      const hours = Math.floor(minutes / 60);
+      const days = Math.floor(hours / 24);
+
+      timeLeft.days = days;
+      timeLeft.hours = hours % 24;
+      timeLeft.minutes = minutes % 60;
+      timeLeft.seconds = seconds % 60;
+    } else {
+      onCountdownFinish();
+    }
+
+    return timeLeft;
+  };
+
+  const [timeLeft, setTimeLeft] = React.useState(calculateTimeLeft());
+
+  useEffect(() => {
+    const timer = setTimeout(() => {
+      setTimeLeft(calculateTimeLeft());
+    }, 1000);
+
+    return () => clearTimeout(timer);
+  });
```

Comment:
We should pass an empty array here (actually, with `calculateTimeLeft` if we won't remove it from the component, so we need to put calculateTimeLeft into a useCallback), because now it runs on each rerender.

Reply:
Done